### PR TITLE
new py-amplpy package

### PR DESCRIPTION
**Pull request by Sangu-Mbekelu**
depends on !35357

### CODE REVIEW

**Comment by Sangu-Mbekelu** [comment]
@spackbot fix style

**Review by tldahlgren — CHANGES_REQUESTED** [review]
As with #35357 , the directory for this package needs to be moved to under `var/spack/repos/builtin/packages`.

Confirmed the sha256.

**Comment by tldahlgren** [comment]
Adding @adamjstewart since this is a `PythonPackage`.

File: var/spack/repos/builtin/py-amplpy/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/py-amplpy/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/py-amplpy/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    and solver interaction is handled directly by AMPL, which leads to great
+    stability and speed; the library just acts as an intermediary, and the
+    added overhead (in terms of memory and CPU usage) depends mostly on how
+    much data is read back from AMPL, the size of the model as such is irrelevant."""
+
+    homepage = "https://ampl.com/"
+    pypi = "amplpy/amplpy-0.8.6.tar.gz"
+
+    version("0.8.6", sha256="ad0945d69f75e7762802bb54849009717fbcf226a6da6f37b539d9534bdcf68d")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/py-amplpy/package.py]: missing build dep on setuptools

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Reconfirmed the sha256.

**Comment by tldahlgren** [comment]
Took the liberty of `Update branch` button to pick up the dependency package.

**Pull request merged**